### PR TITLE
Adding a guide for admins to set up federation

### PR DESCRIPTION
**Pull request by nikhiljindal**
cc @kubernetes/sig-cluster-federation @colhom @mml 

Guide with steps for admin to setup federation

### CODE REVIEW

File: docs/admin/federation/index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+```
+
+By specifying `--cluster=federation-cluster`, we direct the request to federation apiserver.
+You can ensure that the cluster registration was successful by running:
+
+```shell
+$ kubectl get clusters --context=federation-cluster
+NAME       STATUS    VERSION   AGE
+cluster1   Ready               3m
```
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: FYI, I havent been able to reach this state. My cluster remains in "Offline" state. I see the following in my controller manager logs:

```
E0613 03:29:22.404990       1 cluster_client.go:147] Failed to list nodes while getting zone names: the server could not find the requested resource
W0613 03:29:22.405017       1 clustercontroller.go:194] Failed to get zones and region for cluster cluster1: the server could not find the requested resource
```

Am still debugging it

**Comment by nikhiljindal** [comment]
cc @lavalamp who also had suggestions on making this easier for admins to setup

**Comment by nikhiljindal** [comment]
cc @kubernetes/docs @kubernetes/docs-review-team @johndmulhausen 

Can you please confirm that this is the right way to create a doc now?
Do I need to create a placeholder doc in kubernetes repo?

**Comment by nikhiljindal** [comment]
This is part of 1.3 docs

**Comment by lavalamp** [comment]
The placeholders in the main repo are only for auto-generated documents. I thought we copied docs from there to here ~each release. @johndmulhausen would know if the canonical doc should be here or there.

**Comment by bgrant0607** [comment]
@nikhiljindal Needs to go into the release-1.3 branch

**Comment by nikhiljindal** [comment]
@bgrant0607 So I need to move this PR to the core kubernetes repo?

Last time I created a PR there I was told that only auto generated docs are accepted there. All other docs should be edited directly here: https://github.com/kubernetes/kubernetes/pull/26313.

**Comment by rata** [comment]
@nikhiljindal: I guess @bgrant0607 says the branch release-1.3 in this repo.

File: docs/admin/federation/index.md (hunk 1 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: This was a config problem with my setup.
Works great now!

**Comment by nikhiljindal** [comment]
aah thanks @rata Hadnt noticed the branch in this repo :)

@johndmulhausen The [edit instructions](http://kubernetes.io/editdocs/#docs/index.md) are geared towards editing existing docs. We should add instructions for adding docs.

Sent https://github.com/kubernetes/kubernetes.github.io/pull/662 for branch release-1.3.
Closing this.

**Comment by johndmulhausen** [comment]
The edit docs page does include the necessary details on adding pages. See: "Edit Page Titles or Change the Left Navigation" -- definitely will need to add a left nav entry for this file. 

And yes, please ensure release-1.3 is the target for any Kubernetes 1.3 docs.

File: docs/admin/federation/index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Running `federation-up.sh` also creates a new record in your kubeconfig for you
+to be able to talk to federation apiserver. You can view this by running
+`kubectl config view`.
+
+## Registering kubernetes clusters for federation
+Now that we have the federation control plane up and running, we can start registering kubernetes clusters.
+
+First of all, we need to create a secret containing kubeconfig for that kubernetes cluster, which federation control plane will use to talk to that kubernetes cluster.
+For now, we create this secret in the host kubernetes cluster (that hosts federation control plane). When we start supporting secrets in federation control plane, we will create this secret their.
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: s/their/there

File: docs/admin/federation/index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  name: cluster1
+spec:
+  serverAddressByClientCIDRs:
+  - clientCIDR: <client-cidr>
+    serverAddress: <apiserver-address>
+  secretRef:
+    name: <secret-name>
+```
+
+You need to insert the appropriate values for `<client-cidr>`, `<apiserver-address>` and `<secret-name>`.
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: What does the `client-cidr` refer to here?

If I have a cluster from another region I want to add (with its own load balancer), where would I find the CIDR?

File: docs/admin/federation/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+By specifying `--cluster=federation-cluster`, we direct the request to federation apiserver.
+You can ensure that the cluster registration was successful by running:
+
+```shell
+$ kubectl get clusters --context=federation-cluster
+NAME       STATUS    VERSION   AGE
+cluster1   Ready               3m
+```
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: I think we should also include adding a pod/service to the federation CP plane to showcase how load balancing works

**Comment by jamiehannaford** [comment]
One important thing that isn't included here is how DNS works. Right now AWS and GCE are supported, but the docs don't mention how/why DNS is used, or how users can add their own DNS provider.

File: docs/admin/federation/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+You need to insert the appropriate values for `<client-cidr>`, `<apiserver-address>` and `<secret-name>`.
+Assuming your YAML file is located at /cluster1/cluster,yaml, you can run the following command to register this cluster:
+
+```shell
+$ kubectl create -f /cluster1/cluster.yaml --context=federation-cluster
+
+```
+
+By specifying `--cluster=federation-cluster`, we direct the request to federation apiserver.
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: this should be `--context` right?

File: docs/admin/federation/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+This guide explains how to set up a federation cluster. Federation is an alpha feature which lets you control multiple kubernetes clusters.
+[Federation proposal](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/federation.md) has more details.
+
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running kubernetes cluster.
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: I think `have at least one kubernetes clusters` would be better

File: docs/admin/federation/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+If not, then you can head over to the [getting started guides](/docs/getting-started-guides/) to bring up a cluster.
+
+This guide also assumes that you have the kubernetes source code.
+If not, then you can head over to the [download kubernetes guide](/docs/getting-started-guides/binary_release/) to get the source code.
+
+## Setting up federation control plane
+
+### Building and pushing federation control binaries
+
+You can build the federation control plane binaries using the following command:
```
Comment (jamiehannaford) [inline docs/admin/federation/index.md]: Will this be needed? I thought binaries would be built and published for 1.3 release

**Comment by nikhiljindal** [comment]
Thanks for your comments @jamiehannaford 
This PR is now superseded by https://github.com/kubernetes/kubernetes.github.io/pull/662. I will fix your comments there.
Closing this one.

File: docs/admin/federation/index.md (hunk 7 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: Yes. We do not do it right now.
But its a reasonable thing to expect.
Filed https://github.com/kubernetes/kubernetes/issues/27382

Comment (nikhiljindal) [inline docs/admin/federation/index.md]: We are now pushing release binaries. Updated the instructions to make this optional in https://github.com/kubernetes/kubernetes.github.io/pull/662

File: docs/admin/federation/index.md (hunk 5 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: yes, fixed in https://github.com/kubernetes/kubernetes.github.io/pull/662, thx!

File: docs/admin/federation/index.md (hunk 4 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: This guide is geared towards admin use case of setting up federation.
We will have another guide to showcase how developers can use it to created federated services.

File: docs/admin/federation/index.md (hunk 3 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: added more comments in https://github.com/kubernetes/kubernetes.github.io/pull/662

File: docs/admin/federation/index.md (hunk 2 above)
Comment (nikhiljindal) [inline docs/admin/federation/index.md]: fixed in https://github.com/kubernetes/kubernetes.github.io/pull/662, thx!

**Pull request closed without merging**